modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -8875,6 +8921,175 @@ public final void fireEvent(Event event) {
         Event.fireEvent(this, event);
     }
 
+
+    /* *************************************************************************
+     *                                                                         *
+     *                           CSS Transitions                               *
+     *                                                                         *
+     **************************************************************************/
+
+    private List<TransitionTimer<?, ?>> transitionTimers;
+
+    /**
+     * Called by animatable {@link StyleableProperty} implementations in order to register
+     * a running {@link TransitionTimer} with this {@code Node}. This allows the node
+     * to keep track of running timers that are targeting its properties.
+     */
+    private void addTransitionTimer(TransitionTimer<?, ?> timer) {
+        if (transitionTimers == null) {
+            transitionTimers = new ArrayList<>(4);
+        }
+
+        transitionTimers.add(timer);
+    }
+
+    /**
+     * Removes a timer that was previously registered with {@link #addTransitionTimer}.
+     * This method is called by animatable {@link StyleableProperty} implementations
+     * when their {@link TransitionTimer} has completed.
+     */
+    private void removeTransitionTimer(TransitionTimer<?, ?> timer) {
+        if (transitionTimers != null) {
+            transitionTimers.remove(timer);
+        }
+    }
+
+    /**
+     * Finds the transition timer that targets the specified {@code property}.
+     *
+     * @return the transition timer, or {@code null}
+     */
+    private TransitionTimer<?, ?> findTransitionTimer(Property<?> property) {
+        if (transitionTimers == null || transitionTimers.isEmpty()) {
+            return null;
+        }
+
+        for (TransitionTimer<?, ?> timer : transitionTimers) {
+            if (timer.getProperty() == property) {
```

Comment:
minor: this probably works, but I'd still use `equals` here

Reply:
Why would this be a sensible thing to do? I'm quite explicitly comparing the identity of `property`, since I'm interested in finding the one property that I'm looking for, not potentially any property that is in some way "equal" to the property.

For (hopefully) all property implementations, `equals` trivially works because it is not an overridden method and therefore falls back to an identity comparison. What would it even mean for a property to be equal, but not identical to another property?

Reply:
Added a comment.